src/pfe/file-watcher/server/src/projects/projectUtil.ts

```diff
@@ -505,6 +515,17 @@ async function executeBuildScript(operation: Operation, script: string, args: Ar
                             projectInfo.ports.internalDebugPort = containerInfo.internalDebugPort;
                         }
                         logger.logProjectInfo("Found container information: " + JSON.stringify(containerInfo), projectID, projectName);
+
+
+                        const portMappings: any = {};
+                        containerInfo.containerPorts.forEach((port, i) => {
+                            portMappings[port] = containerInfo.hostPorts[i];
+                        });
+
+                        await projectsController.updateProjectInfo(projectID, {
+                            key: "portMappings",
+                            value:  portMappings,
```

Comment:
It looks like we add new mappings here but never remove old ones. If the user changes the exposed port of a project (e.g. from 3000 to 4000) then we will keep the old mapping as well as the new one. (It looks like this causes us to check if the project is up by querying the stale port.)